phac-nml/irida-next
# [ENHC0010020] Delete multiple samples from project samples table

## What does this PR do and why?
This PR adds the ability to delete multiple samples from a project samples table.

In addition to this, other changes related to the changes required for this PR include:
- Fixing Samples and Sample Files Remove functionality to update file checkboxes accordingly
- Updating `infinite_scroll_controller` to use singular/plural forms based on number of selected samples

## Screenshots or screen recordings
[Screencast from 2024-06-19 09:36:05 AM.webm](https://github.com/phac-nml/irida-next/assets/82407232/d9da9e5b-f05e-4dd1-bf24-5574db0d88be)

Multi_status message:
![image](https://github.com/phac-nml/irida-next/assets/82407232/0c206da8-6558-4be8-a4fe-290b5ac5f3b2)

Error:
![image](https://github.com/phac-nml/irida-next/assets/82407232/c4e03812-39d3-40cc-8cbe-ec4d112bd152)


## How to set up and validate locally

Deleting Multiple Samples:
  1. Navigate to a project samples table
  2. Select samples
  Check the following: 
  - The correct number of selected samples appears in the dialog description
  - The correct samples are added to the table listing
  3. Delete the samples. Make sure the expected samples were deleted

Singular/Plural descriptions in sample dialogs:
  1. Select one and multiple samples and open the clone/transfer/delete sample dialogs. Make sure the description uses the singular and plural descriptions accordingly.
  
Test single removal links while the item was selected with a checkbox
- Test remaining checked samples after a removal still behave correctly
  1. Navigate to a project samples table
  2. Check off any number of samples
  3. Use the `Remove` link to remove a sample
  4. Ensure the correct number of selected samples is still displayed. You can test transferring/cloning/deleting samples in this state to ensure the sample selection is still behaving properly
  
- Test "Select All" checkbox and disabled button states after single removal
  1. Select all samples
  2. Remove a single sample using the `Remove` link
  3. Ensure the "Select All" checkbox is still checked
  4. Remove all samples using the `Remove` link
  5. Ensure clone/transfer/delete samples and create export buttons enter the disabled state
  
  Repeat the above for the sample files

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
